proxy: enable client certificate in transport

desimone opened the pull request:
This would enable downstream applications to enforce mutually authenticated connections for TLS connections.  An example app can be found in the [examples repository](https://github.com/pomerium/examples/tree/master/mutual-tls).

**Checklist**:
- [x] updated docs
- [x] unit tests added
- [x] related issues referenced
- [x] updated CHANGELOG.md
- [x] ready for review

desimone commented:
/cc @alagendijk-minddistrict

alagendijk-minddistrict commented:
Could you also add an option for these to be files instead of only the base64 encoded string?

desimone commented:
Thanks @alagendijk-minddistrict & @travisgroth ; 
- added support to add client certs as files in addition to as base64 blobs.
- added support to add custom ca (#179) as files too.  
- updated the example repo to prefer files which are easier to read
- added a few log statements for non-standard routes.
